pkg/generate/code/resource_reference_test.go

```diff
@@ -72,18 +82,17 @@ func Test_ReferenceFieldsValidation_SliceOfReferences(t *testing.T) {
 	// just to test code generation for slices of reference
 	crd := testutil.GetCRDByName(t, g, "VpcLink")
 	require.NotNil(crd)
+
+	field := crd.Fields["SubnetIDs"]
 	expected :=
-		`	if ko.Spec.SecurityGroupRefs != nil && ko.Spec.SecurityGroupIDs != nil {
-		return ackerr.ResourceReferenceAndIDNotSupportedFor("SecurityGroupIDs", "SecurityGroupRefs")
-	}
-	if ko.Spec.SubnetRefs != nil && ko.Spec.SubnetIDs != nil {
+		`	if ko.Spec.SubnetRefs != nil && ko.Spec.SubnetIDs != nil {
```

Comment:
When dealing with arrays best is to check the `len` - since arrays can be non nil still have 0 eleements e.g `subnetRefs: []`

Reply:
True, we do this in other places as well. I copied the old code for this one part, so I'll change it over to the new method.